src/types/Storage/methods/importTransaction.js

```diff
@@ -32,22 +32,32 @@ const importTransaction = function importTransaction(transaction, transactionMet
   let outputIndex = -1;
   const processedAddressesForTx = {};
 
-  // If we already had this transaction locally, we won't add it again,
-  // but we still need to continue processing it as we might have new
-  // address generated (on BIP44 wallets) since the first checkup.
-  if (!transactions[transaction.hash]) {
-    transactions[transaction.hash] = transaction;
-    if (transactionMetadata) {
+  transactions[transaction.hash] = transaction;
+  if (transactionMetadata) {
+    const { height } = transactionMetadata;
+    if (Number.isInteger(height) && height !== 0) {
       transactionsMetadata[transaction.hash] = transactionMetadata;
-      const { height } = transactionMetadata;
+      const mappedTransactionObject = { hash: transaction.hash, ...transactionMetadata };
+
       if (mappedTransactionsHeight[height]) {
```

Comment:
What if height changes due to reorg?

Reply:
How do we get notified about it in the syncWorker ? 
Because, yeah, we would have a duplicate, we need a handler for reorg actually, it does make total sense !

Reply:
We decided to implement reorg handling in a separate PR